lib/product.rb

```diff
@@ -0,0 +1,29 @@
+class Product
+  attr_reader :category,
+              :name,
+              :unit_price
+
+  def initialize (category, name, unit_price, quantity)
+    @category = category
+    @name = name
+    @unit_price = unit_price
+    @quantity = quantity
+  end
+
+  def quantity
+    @quantity.to_i
+  end
+
+  def total_price
+    @unit_price * quantity
+  end
+
+  def is_hoarded?
+    quantity > 99
+  end
```

Comment:
convention is to call the instance variable instead of the method .. that way, you can skip a step in the lookup 😉